File: framework/BaseForm.js
const BaseElement = require('./BaseElement.js')
const logger = require('./logger.js')
const Label = require('./Label.js')

class BaseForm {
  constructor(selector, name) {
    this.selector = selector
    this.name = name
    this.formElement = new Label(selector, name)
  }

  async isVisible() {
    logger.debug(`Checking if ${this.name} is visible`)
    const isVisible = await this.formElement.isDisplayed()
    logger.debug(`Form element ${this.name} visability: ${isVisible}`)
    logger.debug('Checking visability complete')
    return isVisible
  }
}

module.exports = BaseForm

File: src/test/java/framework/utils/BaseForm.java
package framework.utils;

import framework.elements.Label;
import org.openqa.selenium.By;

public class BaseForm {
    protected Label formElement;
    protected String name;
    public BaseForm(By locator, String name){
        this.name = name;
        this.formElement = new Label(locator, name);
    }

    public boolean IsDisplayed(){
        return formElement.IsVisible();
    }
}
